cosmic-panel-bin/src/space/panel_space.rs

```diff
@@ -730,12 +733,43 @@ impl PanelSpace {
         id
     }
 
-    pub fn handle_focus(&mut self) {
-        let (layer_surface, _) = if let Some(layer_surface) = self.layer.as_ref() {
-            (layer_surface, layer_surface.wl_surface())
+    fn start_show_delay(&mut self) {
+        if self.start_show_instant.is_none() {
+            self.start_show_instant = Some(Instant::now());
+        }
+    }
+
+    fn show_delay_done(&mut self, intellihide: bool) -> bool {
+        if intellihide {
+            return true;
+        }
+        if let Some(start_show_instant) = self.start_show_instant {
+            let start_show_duration = match Instant::now().checked_duration_since(start_show_instant) {
+                Some(d) => d,
+                None => return true,
+            };
+            let show_delay_done = start_show_duration >= Duration::from_millis(500);
+            if show_delay_done {
+                self.start_show_instant = None;
+            }
+            show_delay_done
         } else {
-            return;
-        };
+            false
+        }
+    }
+
+    fn reset_show_delay(&mut self) {
+        self.start_show_instant = None;
+    }
+
+    pub fn handle_focus(&mut self) {
+        {
```

Comment:
This is here because of the `self.layer.as_ref` which introduces an immutable borrow on self, I was uncertain if the `wl_surface()` call had some side effects that where import so I left all the code here, if it's not important this could be converted to a simple `is_none` check with an early return.